data/infra/resources/chocolatey_installer.yaml

```diff
@@ -0,0 +1,99 @@
+  chocolatey_installer 'name' do
+    chocolatey_version      String
+    download_url            String
+    ignore_proxy            true, false # default value: false
+    proxy_password          String
+    proxy_url               String
+    proxy_user              String
+    use_native_unzip        true, false # default value: false
+    action                  Symbol # defaults to :install if not specified
+  end
+syntax_properties_list:
+syntax_full_properties_list:
+- "`chocolatey_installer` is the resource."
+- "`name` is the name given to the resource block."
+- "`action` identifies which steps Chef Infra Client will take to bring the node into
+  the desired state."
+- "`chocolatey_version`, `download_url`, `ignore_proxy`, `proxy_password`, `proxy_url`,
+  `proxy_user`, and `use_native_unzip` are the properties available to this resource."
+actions_list:
+  :nothing:
+    shortcode: resources_common_actions_nothing.md
+  :install:
+    markdown: Installs Chocolatey package manager (default)
+  :uninstall:
+    markdown: Uninstall Chocolatey package manager
+  :upgrade:
+    markdown: Upgrades the Chocolatey package manager
+properties_list:
+- property: chocolatey_version
+  ruby_type: String
+  required: false
+  description_list:
+  - markdown: Specifies a target version of Chocolatey to install. By default, the
+      latest stable version is installed. This will use the value in $env:ChocolateyVersion
+      by default, if that environment variable is present. This parameter is ignored
+      if download_url is set.
+- property: download_url
+  ruby_type: String
+  required: false
+  description_list:
+  - markdown: The URL to download Chocolatey from. This sets the value of $env:ChocolateyDownloadUrl
+      and causes the installer to choose an alternate download location. If this is
+      not set, Chocolatey is downloaded from the official Chocolatey community
+      repository. It can also be used for offline installation
+      by providing a path to a Chocolatey.nupkg.
+- property: ignore_proxy
+  ruby_type: true, false
+  required: false
+  default_value: 'false'
+  description_list:
+  - markdown: If set, this overrides any configured proxy, proxy environment
+      variables, or parameters. This is enabled if set
+      to a value other than 'false' or '0'.
+- property: proxy_password
+  ruby_type: String
+  required: false
+  description_list:
+  - markdown: The password to use to build a proxy credential with. Will be consumed
+      by the proxy_credential property if both this property and proxy_user are set.
+- property: proxy_url
+  ruby_type: String
+  required: false
+  description_list:
+  - markdown: Specifies the proxy URL to use during the download.
+- property: proxy_user
+  ruby_type: String
+  required: false
+  description_list:
+  - markdown: The username to use to build a proxy credential with. Will be consumed
+      by the proxy_credential property if both this property and proxy_password are
```

Comment:
Same here.

Reply:
"The user name used to connect to the proxy server with. If this is specified, the proxy password is also expected to be specified. "